meps/mep-cross-references.md

```diff
@@ -0,0 +1,339 @@
+| `` {doc}`my-doc` ``                      | `[](../examples/my-doc.md)`  |
+| `` {download}`my-doc.zip` ``             | `[](my-doc.zip)`             |
+
+[^legacy_hash]: This is backwards compatible, however, now raises a `xref_legacy` warning for old syntax.
+
+All of the above link examples can be easily complemented by both adding Title Link syntax and Reference Link syntax (i.e. `[Explicit *Markdown* text][label]`).
+We have omitted the auto-link syntax from the overview for brevity, they are shown in detail below.
+In all cases, the existing role syntax should continue to work and receive ongoing support from the parser(s).
+
+### Syntax
+
+The parts of the link are `[text](link "title")` with an optional scheme (`[text](scheme:link "title")`). The `"title"` is not modified in our proposed syntax. Auto Link syntax, `<scheme:link>`, requires the scheme to be present, we follow the CommonMark [definition of a scheme](https://spec.commonmark.org/0.30/#scheme).
+
+#### `text`
+
+If the `text` is not included, it will be filled in by the default of the target.
+
+- If the reference target is enumerated this will by default be "Sec 2.3"; "Fig. 1" for enumerated figures; "(5)" for equations; "Thm. 2", "Lemma 3", etc. depending on the target type. This can be customized in project configuration or frontmatter, however, the details of that are out of scope for this MEP.
+- If the reference target is not enumerated, the title will be used. This may be the section header text (including syntax/style) or a figure caption.
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`%s` and `%t`)
+
+- Enumeration (`%s`)
+
+  - Any reference target that is enumerated can reference that number or string with a `%s`.
+  - If the target is enumerated, the default will be the numbered form of the reference (e.g. "Section 2.1.2", "Fig. 3", or "(1)")
+  - If a `%s` is used and the node is not enumerated, the `%s` will be replaced by "??" and a warning raised.
+  - Parsers can optionally choose to support `{number}` which is from Sphinx.
+
+- Title (`%t`)
+  - Any node can include the title of a reference including any styles (e.g. Sections are the section title; Figures and Tables are the caption).
+  - If a `%t` is used and the node does not have an explicit name or title, the node reference label will be used.
+  - Parsers can optionally choose to support `{name}` which is from Sphinx.
+
+#### `link`
+
+The links are defined by a scheme, which can be standard protocols (`http:`, `mailto:`). Here we propose three new schemes, `path`, `project`, and `myst`, which is an extensibility point described by [CommonMark](https://spec.commonmark.org/0.30/#example-598). These schemes are used to indicate that the link should be resolved by MyST specific logic, and follows standard [URI][uri] syntax:
```

Comment:
Should we even mention the `myst` scheme here? It is not explained anywhere else in this document, and maybe gets more into future MEP intersphinx territory

Reply:
Thank you, didn't catch this one on my first pass, this is now removed.